default/python/universe_generation/galaxy.py

```diff
@@ -1,35 +1,297 @@
 import sys
-from random import random, uniform, randint, gauss, choice
-from math import pi, sin, cos, sqrt
+from random import random, uniform, randint, gauss
+from math import pi, sin, cos, acos, sqrt, ceil, floor
+from collections import defaultdict
 
 import freeorion as fo
 import util
 import universe_tables
 
 
-class AdjacencyGrid:
+class AdjacencyGrid(object):
     def __init__(self, universe_width):
+        assert universe_tables.MIN_SYSTEM_SEPARATION < universe_tables.MAX_STARLANE_LENGTH
```

Comment:
AdjacencyGrid only requires MAX_STARLANE_LENGTH > MIN_SYSTEM_SEPARATION so the too_close_to_other_positions() and neighbors() mean different regions.  stitching_positions() below requires that MAX_STARLANE_LENGTH > 2*MIN_SYSTEM_SEPARATION in order to place positions(s) between the nearest neighbors in groups of positions that are too far apart. I reported the most restrictive bound in the PR request.